### PR TITLE
API: Readd `add_docstring` and `add_newdoc` to `np.lib`

### PR DESCRIPTION
**Pull request by mtsokol**
Relevant issue https://github.com/numpy/numpy/issues/24507

Hi @rgommers, 

Reverts a change introduced in https://github.com/numpy/numpy/pull/24538. 
As agreed in the [main namespace design](https://github.com/numpy/numpy/issues/24306#issuecomment-1663564574), `add_docstring` and `add_newdoc` should stay in `np.lib` and be available only from that namespace. These two were unnecessarily removed from there in https://github.com/numpy/numpy/pull/24538. (`np.lib` namespace refactor PR is still TBD, but this needs to be merged to fix SciPy builds)

### CODE REVIEW

File: numpy/_expired_attrs_2_0.py [hunk 1]
```diff
@@ -20,9 +20,9 @@
     "NZERO": "Use `-0.0` instead.",
     "PZERO": "Use `0.0` instead.",
     "add_newdoc": 
-        "It's an internal function and doesn't have a replacement.",
+        "It's now available from `np.lib.add_newdoc`.",
     "add_docstring": 
-        "It's an internal function and doesn't have a replacement.",
+        "It's now available from `np.lib.add_docstring`.",
```
Comment (rgommers) [inline numpy/_expired_attrs_2_0.py:25]: Minor rephrase proposed (use "still ... as"), for both these messages: 
```
It's stilll available as `np.lib.add_docstring`.
```
That hints that it has always been there, not that it recently got moved there.

Comment (mtsokol) [reply numpy/_expired_attrs_2_0.py]: That's right - I changed both lines.

**Review by rgommers — APPROVED** [review]
Thanks for the quick fix @mtsokol. These removals from `numpy.lib` were unintentional, so in this goes. We can revisit the rest later. `add_docstring` looks more questionable and is used less than `add_newdoc`, so I expect we'd clean up at least that one later on.

**Comment by WarrenWeckesser** [comment]
Thanks @mtsokol!

**Pull request merged**